packages/syft/src/syft/custom_worker/builder_k8s.py

```diff
@@ -0,0 +1,318 @@
+# stdlib
+from hashlib import sha256
+import os
+from pathlib import Path
+from typing import Dict
+from typing import Optional
+
+# third party
+import kr8s
+from kr8s.objects import APIObject
+from kr8s.objects import ConfigMap
+from kr8s.objects import Job
+
+# relative
+from .builder_types import BuilderBase
+from .builder_types import ImageBuildResult
+from .builder_types import ImagePushResult
+
+__all__ = ["KubernetesBuilder"]
+
+JOB_COMPLETION_TTL = 60
+
+BUILD_OUTPUT_PVC = "worker-builds"
+KUBERNETES_NAMESPACE = os.getenv("K8S_NAMESPACE", "syft")
+
+
+class InvalidImageDigest(Exception):
+    pass
+
+
+class KubernetesBuilder(BuilderBase):
+    def __init__(self):
+        self.client = kr8s.api(namespace=KUBERNETES_NAMESPACE)
+
+    def build_image(
```

Comment:
NITPICK: We need to add the interface separation in worker_image_service .py `def build` method for branching kubernetes builds, currently it builds using only docker

Reply:
Sorry I didn't get you. The call stack is as follows: `def build()` (worker_image_service.py) -> `docker_build()` (utils.py) -> `CustomWorkerBuilder.build_image()` -> `self.builder.build_image()` which uses a builder determined by runtime.

I think the problem here is the function `docker_build()` in utils.py. Should be called as `worker_build()` and `worker_push()` But i won't address them here

Reply:
Sorry, my bad, with the function name, I assumed , it is only docker build

Reply:
Yep , we could change it later.